interface/therapy_groups/therapy_groups_controllers/therapy_groups_controller.php

```diff
@@ -0,0 +1,386 @@
+
+        //Insert static arrays to send to view.
+        $data['statuses'] = SELF::$statuses;
+        $data['group_types'] = SELF::$group_types;
+        $data['group_participation'] = SELF::$group_participation;
+        $data['counselors'] = $this->prepareCounselorsList($counselors);
+
+        //Send groups array to view.
+        $this->loadView('listGroups', $data);
+    }
+
+    /**
+     * Prepares the therapy group list that will be sent to view.
+     * @param $therapy_groups
+     * @param $counselors
+     * @return array
+     */
+    private function prepareGroups($therapy_groups, $counselors){
+
+        $new_array = array();
+        $users_model = $this->loadModel('Users');
+
+        //Insert groups into a new array and shorten notes for preview in list
+        foreach ($therapy_groups as $therapy_group) {
+            $gid = $therapy_group['group_id'];
+            $new_array[$gid] = $therapy_group;
+            $new_array[$gid]['group_notes'] = $this->shortenNotes($therapy_group['group_notes']);
+            $new_array[$gid]['counselors'] = array();
+        }
+
+        //Insert the counselors into their groups in new array.
+        foreach ($counselors as $counselor){
+            $group_id_of_counselor = $counselor['group_id'];
+            $counselor_id = $counselor['user_id'];
+            $counselor_name = $users_model->getUserNameById($counselor_id);
+            if(is_array($new_array[$group_id_of_counselor])) {
+                array_push($new_array[$group_id_of_counselor]['counselors'], $counselor_name);
+            }
+        }
+
+        return $new_array;
+
+    }
+
+    private function shortenNotes($notes){
+
+        $length = strlen($notes);
+        if($length > $this->notes_preview_proper_length){
+            $notes = mb_substr($notes,0,50).'...';
+        }
+        return $notes;
+    }
+
+    /**
+     * Returns a list of counselors without duplicates.
+     * @param $counselors
+     * @return array
+     */
+    private function prepareCounselorsList($counselors){
+
+        $new_array = array();
+        $users_model = $this->loadModel('Users');
+
+        foreach ($counselors as $counselor){
+            $counselor_id = $counselor['user_id'];
+            $counselor_name = $users_model->getUserNameById($counselor_id);
+            $new_array[$counselor_id] = $counselor_name;
+        }
+
+        return $new_array;
+    }
+
+    /**
+     * Change group status to 'deleted'. Can be done only if group has no encounters.
+     * @param $group_id
+     * @return array
+     */
+    private function deleteGroup($group_id){
+
+        $response = array();
+
+        //If group has encounters cannot delete the group.
+        $group_has_encounters = $this->checkIfHasApptOrEncounter($group_id);
+        if($group_has_encounters){
+            $response['success'] = 0;
+            $response['message'] = xl("Deletion failed because group has appointments or encounters");
```

Comment:
via html and it's been escaped there: 
/home/shaharzi/projects/c_n/openemr/interface/therapy_groups/therapy_groups_views/listGroups.php
```
   <?php if($deletion_try == 1 && $deletion_response['success'] == 0) :?>
        <div class="row">
            <div class="col-md-6 col-md-offset-3">
                <div class="alert alert-danger text-center">
                    <p class="failed_message"><?php echo xlt($deletion_response['message']);?></p>
                </div>
            </div>
        </div>
    <?php endif ?>

````